homeassistant/components/vizio/media_player.py

```diff
@@ -84,22 +67,24 @@
         host,
         name,
         token,
-        device_type,
+        VIZIO_DEVICE_CLASSES[device_class],
         session=async_get_clientsession(hass, False),
     )
 
     if not await device.can_connect():
         fail_auth_msg = ""
         if token:
-            fail_auth_msg = ", auth token is correct"
+            fail_auth_msg = "and auth token '{token}' are correct."
+        else:
+            fail_auth_msg = "is correct."
         _LOGGER.error(
-            "Failed to set up Vizio platform, please check if host "
-            "is valid and available, device type is correct%s",
+            "Failed to connect to Vizio device, please check if host '{host}'"
+            "is valid and available. Also check if device class '{device_class}' %s",
             fail_auth_msg,
         )
-        raise PlatformNotReady
+        raise ConfigEntryNotReady
```

Comment:
In the platform set up we still use `PlatformNotReady` even from within `async_setup_entry`.

It only works to raise `ConfigEntryNotReady` from within the main `async_setup_entry` in the component.